lib/src/geometry/collision_detection.dart

```diff
@@ -1,18 +1,32 @@
 import '../../extensions.dart';
 import '../components/mixins/collidable.dart';
 
+void aloneCollisionDetection(Collidable collidableX, Collidable collidableY) {
+  final points = intersections(collidableX, collidableY);
+  if (points.isNotEmpty) {
+    collidableX.onCollision(points, collidableY);
+    collidableY.onCollision(points, collidableX);
+  }
+}
+
 /// Check whether any [Collidable] in [collidables] collide with each other and
 /// call their onCollision methods accordingly.
-void collisionDetection(List<Collidable> collidables) {
-  for (var x = 0; x < collidables.length; x++) {
-    final collidableX = collidables[x];
-    for (var y = x + 1; y < collidables.length; y++) {
-      final collidableY = collidables[y];
-      final points = intersections(collidableX, collidableY);
-      if (points.isNotEmpty) {
-        collidableX.onCollision(points, collidableY);
-        collidableY.onCollision(points, collidableX);
-      }
+void collisionsDetection(
+  List<Collidable> activeCollidables,
+  List<Collidable> noActiveCollidables,
+) {
+  for (var x = 0; x < activeCollidables.length; x++) {
```

Comment:
I don't think we need to maintain two lists, just skipping the element in this outer loop if it is not active should be a very cheap operation.

Reply:
No, because then we have to create new lists on every iteration, that is too expensive.
Just keep the original `collidables` list and check if each element is active in the loop, so just like your example, but with only one list. And then we don't have to update the other two lists in each tick to check if a collidable has changed its active state.

Reply:
I think I know, your idea, he will be changed later, thanks

Reply:
I may use

```
   for (var x = 0; x < allActiveCollidables.length; x++) {
    if(!allActiveCollidables[x].activeCollidable){
         continue;
    }
    for(var y=0;y<allActiveCollidables.length,y++)

```

**Note that y = 0**

Otherwise, these codes may skip some collision detection. If you have other ideas, can you use some codes to prompt me

Reply:
The problem with those loops is that it will make the solution more inefficient when you have less inactive collidables than active ones, which I'm guessing will be the case for most people. I will try to think of a better solution.
Will you allow me to push to your branch? Then I can start writing the docs too.

Reply:
> 这些循环的问题在于，当您的非活动可折叠对象少于活动的可折叠对象时，它将使解决方案的效率降低，我想大多数人都会遇到这种情况。我将尝试考虑一个更好的解决方案。
> 你能让我推到你的分店吗？然后，我也可以开始编写文档了。

OK, I've given you permission,In some places, I believe better improvements can be made.

Reply:
Thanks!